### DISCUSSION TITLE
Why use the OpenZep for a simple modifier?

### DISCUSSION THREAD

**Discussion by Dave777** [smartcontractkit/full-blockchain-solidity-course-py#948] [General]
Hi,
Like many of us here I am working through the tutorial.

There seems to be a love for linking out to code even when it appears more
difficult than simple including the code in the contract.

For example at 6:34:20 it is suggested to use OpenZeppelin instead of just coding the modifier
directly into the contract.

Surely this is just obscuring the code and creating another dependency which could change or break.
Isn't it quicker easier and safer just to code in the modifier?

Thought I would share ... what do you think?

**Comment by n4n0b1t3** [comment]
The course manages to introduce an astonishing range of important tools and libraries that occur in the web3 context and also actively integrates them into the programming examples. Due to the nature of this being a beginners course those examples are very simple. In the end, if you were to use only a single and easy-to-implement function of the OpenZeppelin library in your source code, then I agree with you, it would be overkill to import those libs. But that is rarely the case. OpenZeppelin is a good example of a mature library used by thousands of developers in production. With it you buy the following advantages as a developer:

- The peer-reviewed and by life projects widely adopted code base significantly increases security. 
- Instead of rewriting existing functionality, you can focus on the unique selling points of your app.
- The time to market for your app can be significantly reduced.

These are three good points why you should use OpenZeppelin or other open-source libraries. Here in the course, this is just an exercise.

**Comment by Dave777** [comment]
Thanks for your comment.
Very reasonable and well articulated.

**Comment by cromewar** [comment]
Yeah @n4n0b1t3 is right, we can ended create all the modifiers by ourselves, but using Openzeppelin is just the way to make your life easier and save a lot of time at developing. I just want to make an annotation here:

Is not likely your code/project to broke due to the changes or updates on the library as we setup the specific version we want to use on `brownie-cconfig.yaml`.

Best regards, Cromewar.
